substrate/frame/support/src/lib.rs

```diff
@@ -2262,6 +2262,234 @@ pub mod pallet_macros {
+	///
+	/// ## Example:
+	///
+	/// ```
+	/// #[frame_support::pallet]
+	/// mod pallet {
+	///     use frame_support::pallet_prelude::*;
+	///
+	///     #[pallet::config]
+	///     pub trait Config: frame_system::Config { }
+	///
+	///     #[pallet::pallet]
+	///     pub struct Pallet<T>(_);
+	///
+	/// 	#[pallet::call]
+	/// 	impl<T: Config> Pallet<T> {
+	/// 		pub fn do_stuff(_origin: OriginFor<T>, arg: u32) -> DispatchResult {
+	/// 			unimplemented!()
+	/// 		}
+	/// 	}
+	/// }
+	/// ```
+	///
+	/// TODO: Once we have default pallet config, it would also be easy to create types in the
+	/// example that implement `Config`.
+	pub use frame_support_procedural::call;
+
+	/// Declares a type alias as a storage item. Storage items are pointers to data stored on-chain
+	/// (the *blockchain state*), under a specific key. The exact key is dependent on the type of
+	/// the storage.
+	/// 
+	/// > Hypothetically, one can directly manipulate the state via
+	/// > [`sp_io::storage`](sp_io::storage). However, this is an advance usage and is not
+	/// > recommended.
+	/// 
+	/// ## Storage Types
+	/// 
+	/// The following storage types are supported by the FRAME macros.
+	///
+	/// * [`StorageValue`](frame_support::storage::types::StorageValue)
+	/// * [`StorageMap`](frame_support::storage::types::StorageMap)
+	/// * [`CountedStorageMap`](frame_support::storage::types::CountedStorageMap)
+	/// * [`StorageDoubleMap`](frame_support::storage::types::StorageDoubleMap)
+	/// * [`StorageNMap`](frame_support::storage::types::StorageNMap)
+	/// * [`CountedStorageNMap`](frame_support::storage::types::CountedStorageNMap)
+	///
+	/// The FRAME macros always generate a type alias for one of these types, as indicated by the
+	/// syntax, for example: `type Foo = StorageValue<..>`. For specific information about each
+	/// storage type, refer to the documentation of the respective type.
+	///
+	/// ### Example:
+	/// 
+	/// ```
+	/// #[frame_support::pallet]
+	/// mod pallet {
+	///     # use frame_support::pallet_prelude::*;
+	///     # #[pallet::config]
+	///     # pub trait Config: frame_system::Config {}
+	///     # #[pallet::pallet]
+	///     # pub struct Pallet<T>(_);
+	/// 	#[pallet::storage]
+	/// 	type Foo<T> = StorageValue<_, u32>;
+	///
+	/// 	#[pallet::storage]
+	/// 	type Bar<T> = StorageMap<_, Blake2_128Concat, u32, u32>;
+	///
+	/// 	#[pallet::storage]
+	/// 	type Baz<T> = CountedStorageMap<_, Blake2_128Concat, u32, u32>;
+	/// }
+	/// ```
+	/// 
+	/// ## Related Macros
+	/// 
+	/// The following macros can be used in conjunction with the storage macro:
+	///
+	/// * [`macro@getter`]: Creates a custom getter function.
+	/// * [`macro@storage_prefix`]: Overrides the default prefix of the storage item.
+	/// * [`macro@unbounded`]: Declares the storage item as unbounded.
+	///
+	/// ## Common Details to All Storage Types
+	///
+	/// The following details are relevant to all of the aforementioned storage types.
```

Comment:
I think this could be a great section with more elaboration of what the API needs to be consumed.  For example, a high-level explanation of all the factors that using of storage type would fit well here.

If you agree, I can write a more elaborate section.

```suggestion
	/// ## Storage Type Usage
	///
	/// The following details are relevant to all of the aforementioned storage types.  Storage types require the 
    /// following items ... (bullet list of all arguments / generic params here)
```

Reply:
thanks, I included a section like this :)